Assets/MixedRealityToolkit-SDK/Features/Input/Scripts/PointerClickHandler.cs

```diff
@@ -0,0 +1,95 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Internal.Definitions.InputSystem;
+using Microsoft.MixedReality.Toolkit.Internal.EventDatum.Input;
+using Microsoft.MixedReality.Toolkit.Internal.Interfaces.InputSystem;
+using Microsoft.MixedReality.Toolkit.Internal.Interfaces.InputSystem.Handlers;
+using Microsoft.MixedReality.Toolkit.Internal.Managers;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.SDK.Input
+{
+    /// <summary>
+    /// This component handles pointer clicks from all types of input sources.<para/>
+    /// i.e. a primary mouse button click, motion controller selection press, or hand tap.
+    /// </summary>
+    public class PointerClickHandler : MonoBehaviour, IMixedRealityPointerHandler
+    {
+        private IMixedRealityInputSystem inputSystem = null;
+        private IMixedRealityInputSystem InputSystem => inputSystem ?? (inputSystem = MixedRealityManager.Instance.GetManager<IMixedRealityInputSystem>());
+
+        [SerializeField]
+        [Tooltip("Is Gaze required for the action to raise the event?")]
+        private bool isGazeRequired = false;
+
+        [SerializeField]
+        [Tooltip("The input actions to be recognized on pointer up.")]
+        private InputActionEventPair onPointerUpActionEvents;
+
+        [SerializeField]
+        [Tooltip("The input actions to be recognized on pointer down.")]
+        private InputActionEventPair onPointerDownActionEvents;
+
+        [SerializeField]
+        [Tooltip("The input actions to be recognized on pointer clicked.")]
+        private InputActionEventPair onPointerClickedActionEvents;
```

Comment:
I'm guessing this will need it's own property drawer to render properly in the inspector.  Not entirely sure I like this approach, but let's see where it leads us.

Reply:
It doesn't need a property drawer

Reply:
So you only expect this to be accessible via code?  Or do you have a screenshot of how this field looks like in the editor?
It's not simpler if the user can't use the editor too.

Reply:
@SimonDarksideJ, these look to be ONLY accessible via the inspector.
@StephenHodgson, should these be accessible via script?

Reply:
The opposite actually.  The events will be accessible and setup in the editor only.

![image](https://user-images.githubusercontent.com/13334553/42894385-8f89b43e-8a6b-11e8-809a-cae824b4e45d.png)

Reply:
>  should these be accessible via script?

That is correct